src/operators/RecoverWith.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * Created by srijan02420 on 12/10/18.
+ */
+
+import {CompleteMixin, Virgin} from '../internal/Mixins'
+import {IObservable} from '../internal/Observable'
+import {IObserver} from '../internal/Observer'
+import {ISubscription} from '../internal/Subscription'
+import {curry} from '../internal/Utils'
+import {IScheduler} from '../schedulers/Scheduler'
+
+export type TSource<T> = IObservable<T>
+export type TResult<R> = IObservable<R>
+export type TMapper<T> = (err: Error) => T
```

Comment:
T should be an observable here. Once the values are recovered there is no way to detect if the returned observable carries a value or an error.

Reply:
@tusharmath you mean source should be observable of observable?